File: app/src/main/java/com/rungenes/firebasevisualsapplication/ViewHolder.java
package com.rungenes.firebasevisualsapplication;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

public class ViewHolder extends RecyclerView.ViewHolder {


    View view;

    public ViewHolder(@NonNull View itemView) {

        super(itemView);
        view = itemView;

        //on item click

        itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if (mClickListener!=null) {

                    int position = getAdapterPosition();
                    if (position != RecyclerView.NO_POSITION) {

                        mClickListener.onItemClick(view, position);
                    }
                }

            }
        });
        itemView.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View view) {



                return true;
            }
        });
    }

    //setting details to recyclerview row

    public void setDetails(Context context, String title, String image, String description) {
        //views
        TextView textViewTitle, textViewDescription;
        ImageView imageViewRow;

        textViewTitle = view.findViewById(R.id.textViewTitle);
        textViewDescription = view.findViewById(R.id.textViewDescription);
        imageViewRow = view.findViewById(R.id.imageViewRow);

 /*       ModelClass modelClass = new ModelClass();

        textViewTitle.setText(modelClass.getTitle());
        textViewDescription.setText(modelClass.getDescription());

        Picasso.get().load(modelClass.getImage()).into(imageViewRow);*/

        textViewTitle.setText(title);
        Picasso.get().load(image).into(imageViewRow);
        textViewDescription.setText(description);

    }

    //interface to send callbacks

    private ViewHolder.ClickListener mClickListener;

    public interface ClickListener{
        void onItemClick(View view, int position);
        void onItemLongClick(View view, int position);

    }

    public void setOnClickListener(ViewHolder.ClickListener clickListener){
        mClickListener = clickListener;
    }


}

File: app/src/main/java/com/rungenes/firebasevisualsapplication/ViewHolder.kt
package com.rungenes.firebasevisualsapplication

import android.content.Context
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import android.widget.TextView
import com.rungenes.firebasevisualsapplication.R
import com.squareup.picasso.Picasso
import com.rungenes.firebasevisualsapplication.ViewHolder.ClickListener
import android.view.View.OnLongClickListener
import android.widget.ImageView

class ViewHolder(var view: View) : RecyclerView.ViewHolder(view) {
    //setting details to recyclerview row
    fun setDetails(context: Context?, title: String?, image: String?, description: String?) {
        //views
        val textViewTitle: TextView = view.findViewById(R.id.textViewTitle)
        val textViewDescription: TextView = view.findViewById(R.id.textViewDescription)
        val imageViewRow: ImageView = view.findViewById(R.id.imageViewRow)

        /*       ModelClass modelClass = new ModelClass();

        textViewTitle.setText(modelClass.getTitle());
        textViewDescription.setText(modelClass.getDescription());

        Picasso.get().load(modelClass.getImage()).into(imageViewRow);*/textViewTitle.text = title
        Picasso.get().load(image).into(imageViewRow)
        textViewDescription.text = description
    }

    //interface to send callbacks
    private var mClickListener: ClickListener? = null

    interface ClickListener {
        fun onItemClick(view: View?, position: Int)
        fun onItemLongClick(view: View?, position: Int)
    }

    fun setOnClickListener(clickListener: ClickListener?) {
        mClickListener = clickListener
    }

    init {

        //on item click
        itemView.setOnClickListener { view ->
            if (mClickListener != null) {
                val position = adapterPosition
                if (position != RecyclerView.NO_POSITION) {
                    mClickListener!!.onItemClick(view, position)
                }
            }
        }
        itemView.setOnLongClickListener { view ->
            if (mClickListener != null) {
                val position = adapterPosition
                if (position != RecyclerView.NO_POSITION) {
                    mClickListener!!.onItemLongClick(view, position)
                }
            }
            true
        }
    }
}